save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/ManageUserRoleCard.kt

```diff
@@ -282,19 +281,24 @@ fun manageUserRoleCardComponent() = fc<ManageUserRoleCardProps> { props ->
                             updatePermissions()
                         }
                         attrs.id = "role-$userIndex"
-                        for (role in Role.values()) {
-                            if (role != Role.NONE && (role.priority < selfRole.priority || role == userRole)) {
+                        Role.values()
+                            .filter { it != Role.NONE }
+                            .filter { it != Role.SUPER_ADMIN }
+                            .filter { selfRole == Role.OWNER || it.isLowerThan(selfRole) || userRole == it }
```

Comment:
I suggest to extract this to a function `rolesAssignableBy(selfRole)`

Reply:
Finally done!